googleapiclient/discovery_cache/file_cache.py

```diff
@@ -0,0 +1,89 @@
+# Copyright 2014 Google Inc. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""File based cache for the discovery document."""
```

Comment:
I'm still unclear as to the use case for this, and the module doc string isn't adding enough details. Doesn't the use of tempfile preclude cross-process cache sharing as a use case? Is this intended just for a single process?

Also, it looks like the whole cache gets read on every get and set call and written on every set call - doesn't that seem like it will have trouble scaling? Remind me again of the reason for using a file rather than just caching in memory?